### ISSUE TITLE
Improper implementation of software and potentially hardware interrupts

### ISSUE THREAD

**Issue by JeremyWildsmith** [HerobrinesArmy/DevCPU#6]
## Improper implementation of software and potentially hardware interrupts

Version 0.0.0: devcpu.win32.win32.x86_64
## Recreating issue

This code sample does not appear to execute correctly, explanation follows:

```
ias testHandler
iaq 0
set a, 0
int 0
sub pc, 1

:testHandler
    set a, 0xFFFF
    rfi a
```
## Explanation & Work-around

One would expect that PC be equal to approximately 5 or 7 after a couple CPU cycles have been run. However after executing this in DevCPU, the PC seems to jump around to 'random' values and never stabilizes.

After further review, it seems that once the interrupt handler is invoked, the return address is not pushed to the stack, since pushing a return address before invoking the software interrupt seems to illustrate the desired effect:

```
ias testHandler
iaq 0
set a, 0
set push, return
int 0
return:
sub pc, 1

:testHandler
    set a, 0xFFFF
    rfi a
```
## Proposed Solution

As per the spec on dcpu.com:

"When IA is set to something other than 0, interrupts triggered on the DCPU-16
will turn on interrupt queueing, push PC to the stack, followed by pushing A to
the stack, then set the PC to IA, and A to the interrupt message."

Implementation of the second stage (pushing pc to the stack) is lacking and should be implemented.
## Additional Notes

This has only been tested with software interrupts, though I encourage you also ensure this is not the case with hardware interrupts as well.

**Comment by Lucus16** [comment]
Fixed in pull request #7

**Comment by HerobrinesArmy** [comment]
Lucus found a bug in RFI that was causing this behavior (prefixed increments of SP instead of postfixed) and his fix for it has been merged into the master. Thanks for the detailed report!
